apps/web/src/app/[locale]/challenge/[id]/left-wrapper.tsx

```diff
@@ -81,62 +91,91 @@ export function LeftWrapper({ challengeId, children, expandPanel, isDesktop }: P
     return 'description';
   }, [pathname]);
 
+  // Hide the enrolled track when in collapsed mobile view.
+  const isTrackEnabled = featureFlags?.enableInChallengeTrack;
```

Comment:
Also if the session is defined?

Reply:
That’s probably a good check here. The enrolled `track` will be null without a user and `<ChallengeTrackNavigation />` self-hides without track data, but your suggestion is good. I agree, I prefer removing an element from the tree instead of mounting an element that self-hides.

I’ll make that update 👍